shared/src/price_estimate.rs

```diff
@@ -166,13 +169,27 @@ impl UniswapPriceEstimator {
             |_, path, pools| estimate_spot_price(path, pools),
         )
         .await
-        .and_then(|(path, price)| {
-            Ok((
-                path,
-                big_rational_to_float(price)
-                    .ok_or_else(|| anyhow!("Cannot convert price ratio to float"))?,
-            ))
-        })
+    }
+
+    // Returns a vector of (rational) prices for the given tokens denominated
+    // in denominator_token or an error in case there is an error computing any
+    // of the prices in the vector.
+    pub async fn best_execution_spot_prices(
+        &self,
+        tokens: &[H160],
+        denominator_token: H160,
+    ) -> Result<Vec<BigRational>> {
+        let res = join_all(tokens.iter().map(|token| async move {
+            if *token != denominator_token {
+                self.best_execution_spot_price(*token, denominator_token)
+                    .await
+            } else {
+                Ok((vec![*token, *token], BigRational::from_integer(1.into())))
```

Comment:
Ubernit: Instead of returning a tuple of Vec<H160>, BigRational here, you could only return the BigRational (and in the if branch use `.map(|t| t.1)`.

I find it slightly awkward to temporarily create an invalid path.

Reply:
Thanks, I didn't realize that introducing the sync block allowed for a much simpler implementation.